File: lesson_22_java/src/main/java/com/lesson22java/services/ProductService.java
package com.lesson22java.services;

import com.lesson22java.entities.Product;
import com.lesson22java.repositories.ProductRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor
public class ProductService {
    private ProductRepository productRepository;

    public Product create(Product product) {
        return productRepository.save(product);
    }

    public Iterable<Product> getAll() {
        return productRepository.findAll();
    }

    public Product getOne(Long id) {
        return productRepository.findById(id).orElseThrow(() -> new RuntimeException("No such product"));
    }

    public Product edit(Product product) {
        return productRepository.save(product);
    }

    public void delete(Long id) {
        productRepository.deleteById(id);
    }
}

File: lesson_22/src/main/kotlin/com/lesson22/services/ProductService.kt
package com.lesson22.services

import com.lesson22.entities.Product
import com.lesson22.repositories.CustomerRepository
import com.lesson22.repositories.ProductRepository
import org.springframework.stereotype.Service

@Service
class ProductService (val productRepository: ProductRepository){
    fun getAll() = productRepository.findAll()
    fun getById(id: Long) = productRepository.findById(id)
    fun create(product: Product) = productRepository.save(product)
    fun delete(id: Long) = productRepository.deleteById(id)
    fun edit(product: Product) = productRepository.save(product)
}